src/main/groovy/uk/gov/hmrc/jenkinsjobs/domain/step/Steps.groovy

```diff
@@ -56,12 +56,19 @@ class Steps {
                   |
                   |mkdir dist && cd dist
                   |
-                  |for version in \$(find ../assets/public/* -type d -d 0); do
+                  |for version in \$(find ../assets/public/* -maxdepth 0 -type d); do
                   |  npm pack \$version
                   |done
                   |
+                  |cat >\${WORKSPACE}/.npmrc <<EOL
+                  |registry=https://api.bintray.com/npm/hmrc/npm-release-candidates
+                  |_auth=\${BINTRAY_CREDENTIALS}
```

Comment:
Does this overwrite/ append to the content of the file? If so do we put the original content back at the end?

Also how do this affect jobs which may be running at the same time which rely on the content of the file being the original content? Presumably they would fail?

Reply:
This creates a new .npmrc file in the project directory (i.e. the workspace). Due to the way npm looks for config this will take precedence over .npmrc files in other locations such as the jenkin's user $HOME path. The other `npm` commands all interact with the default public registry, and as such do not require any npmrc config to be present. By injecting it into the workspace just before the `npm publish` command runs, we ensure that it only impacts that one call. Make sense?

Reply:
Aah of course, totally missed that it's being written to the workspace. Nice!